regexec.c

```diff
@@ -235,6 +235,10 @@ S_regcppush(pTHX_ const regexp *rex, I32 parenfloor, U32 maxopenparen _pDEPTH)
     I32 p;
     DECLARE_AND_GET_RE_DEBUG_FLAGS;
 
+    /* static assert that offs struc size is not less than stack elem size */
+
+    sizeof(char[sizeof(*rex->offs) >= sizeof(*PL_savestack) ? 1 : -1]);
```

Comment:
See `STATIC_ASSERT_DECL` and `STATIC_ASSERT_STMT` in `perl.h`

This should ideally be used at the point we rely on it for the calculation in `regcppop`, something like:
```
     /* calculate number of offs/capture groups stored */
+    STATIC_ASSERT_STMT(sizeof(*rex->offs) >= sizeof(*PL_savestack));
     i = (i * sizeof(*PL_savestack)) / sizeof(*rex->offs);
```

Reply:
This still needs to move to reccppop

Reply:
Not sure if I marked this unresolved last time, but it still needs to move.

Reply:
It's a static assert I can move it if you like.

Reply:
just curious, can it be made into a compile time assert somehow?

Reply:
> just curious, can it be made into a compile time assert somehow?

Yes, per my comment above: "See STATIC_ASSERT_DECL and STATIC_ASSERT_STMT in perl.h".